Source/Dafny/AST/DafnyAst.cs

```diff
@@ -13,14 +13,92 @@
 using System.Diagnostics.Contracts;
 using System.Numerics;
 using System.Linq;
-using Microsoft.Boogie;
 using System.Diagnostics;
 using System.Threading;
+using Microsoft.Boogie;
 
 namespace Microsoft.Dafny {
   [System.AttributeUsage(System.AttributeTargets.Field)]
   public class FilledInDuringResolutionAttribute : System.Attribute { }
 
+  public interface IToken : Microsoft.Boogie.IToken {
+    // Inherited for now.
+    /*
+    int kind { get; set; }
+    string filename { get; set; }
+    int pos { get; set; }
+    int col { get; set; }
+    int line { get; set; }
+    string val { get; set; }
+    bool IsValid { get; }*/
+    string leadingTrivia { get; set; }
+    string trailingTrivia { get; set; }
+  }
+
+  public record Token : IToken {
+    private int _kind;
+    private string _filename;
+    private int _pos;
+    private int _col;
+    private int _line;
+    private string _val;
+    private string _leadingTrivia;
+    private string _trailingTrivia;
+    public Token next;
+    public static readonly IToken NoToken = (IToken)new Token();
+
+    public Token() => this._val = "anything so that it is nonnull";
+
+    public Token(int linenum, int colnum) {
+      this._line = linenum;
+      this._col = colnum;
+      this._val = "anything so that it is nonnull";
+    }
+
+    public int kind {
```

Comment:
Why not use auto-properties?

Reply:
Good point. I was just copying Boogie's code. Updated.